collectors/ebpf.plugin/ebpf_socket.c

```diff
@@ -289,30 +308,23 @@ static void ebpf_socket_send_data(ebpf_module_t *em)
 
     // We read bytes from function arguments, but bandiwdth is given in bits,
     // so we need to multiply by 8 to convert for the final value.
-    write_count_chart(
-      NETDATA_TCP_FUNCTION_COUNT, NETDATA_EBPF_FAMILY, socket_publish_aggregated, 3);
-    write_io_chart(
-        NETDATA_TCP_FUNCTION_BITS, NETDATA_EBPF_FAMILY, socket_id_names[0], common_tcp.write*8/1000,
-        socket_id_names[1], common_tcp.read*8/1000);
+    write_count_chart(NETDATA_TCP_FUNCTION_COUNT, NETDATA_EBPF_IP_FAMILY, socket_publish_aggregated, 3);
+    write_io_chart(NETDATA_TCP_FUNCTION_BITS, NETDATA_EBPF_IP_FAMILY, socket_id_names[0],
+                   common_tcp.write*8/1000, socket_id_names[1], common_tcp.read*8/1000);
     if (em->mode < MODE_ENTRY) {
-        write_err_chart(
-          NETDATA_TCP_FUNCTION_ERROR, NETDATA_EBPF_FAMILY, socket_publish_aggregated, 2);
-    }
-    write_count_chart(
-        NETDATA_TCP_RETRANSMIT, NETDATA_EBPF_FAMILY, &socket_publish_aggregated[NETDATA_IDX_TCP_RETRANSMIT],
-        1);
-
-    write_count_chart(
-        NETDATA_UDP_FUNCTION_COUNT, NETDATA_EBPF_FAMILY, &socket_publish_aggregated[NETDATA_IDX_UDP_RECVBUF],
-        2);
-    write_io_chart(
-        NETDATA_UDP_FUNCTION_BITS, NETDATA_EBPF_FAMILY,
-        socket_id_names[3],(long long)common_udp.write*8/100,
-        socket_id_names[4], (long long)common_udp.read*8/1000);
+        write_err_chart(NETDATA_TCP_FUNCTION_ERROR, NETDATA_EBPF_IP_FAMILY, socket_publish_aggregated, 2);
+    }
+    write_count_chart(NETDATA_TCP_RETRANSMIT, NETDATA_EBPF_IP_FAMILY,
+                      &socket_publish_aggregated[NETDATA_IDX_TCP_RETRANSMIT],1);
+
+    write_count_chart(NETDATA_UDP_FUNCTION_COUNT, NETDATA_EBPF_IP_FAMILY,
+                      &socket_publish_aggregated[NETDATA_IDX_UDP_RECVBUF],2);
+    write_io_chart(NETDATA_UDP_FUNCTION_BITS, NETDATA_EBPF_IP_FAMILY,
+                   socket_id_names[3],(long long)common_udp.write*8/100,
+                   socket_id_names[4], (long long)common_udp.read*8/1000);
     if (em->mode < MODE_ENTRY) {
-        write_err_chart(
-            NETDATA_UDP_FUNCTION_ERROR, NETDATA_EBPF_FAMILY, &socket_publish_aggregated[NETDATA_UDP_START],
-            2);
+        write_err_chart(NETDATA_UDP_FUNCTION_ERROR, NETDATA_EBPF_IP_FAMILY,
+                        &socket_publish_aggregated[NETDATA_UDP_START], 2);
```

Comment:
I think this logic explains why @vlvkobal & I cannot see the udp/tcp function error charts.

We need to update `/etc/netdata/ebpf.d.conf`:

```diff
-    ebpf load mode = entry
+    ebpf load mode = return
```

We could put this in the PR description.

Reply:
There is
```
[global]
    ebpf load mode = return
```
in my `ebpf.d.conf`

Reply:
Hmmm, not sure why it doesn't show then 🤔

Reply:
I will retest on Manjaro to try replicate.

Reply:
@vlvkobal I tested on Manjaro and I had charts, do you have the file `/etc/netdata/ebpf.d/network.conf`? I am thinking that inside it we have `ebpf load mode = entry`.

Reply:
I found the problem after @vlvkobal gives me his config files. I already pushed a commit for it.